libsolidity/formal/SymbolicTypes.cpp

```diff
@@ -233,14 +233,18 @@ pair<bool, shared_ptr<SymbolicVariable>> newSymbolicVariable(
 	bool abstract = false;
 	shared_ptr<SymbolicVariable> var;
 	frontend::Type const* type = &_type;
-	if (!isSupportedTypeDeclaration(_type))
+
+	if (auto userType = dynamic_cast<UserDefinedValueType const*>(type))
+		type = &userType->underlyingType();
```

Comment:
Maybe it's easier to implement this by returning `newSymbolicVariable(&userType->underlyingType(), _uniqueName, _context)`?

Reply:
I'm not sure that'd be easier. `newSymbolicVariable` is used in 4 places, so we'd need this logic in all those places. I'd prefer to centralize that here.